pkg/controllers/servicesync/servicesync.go

```diff
@@ -195,6 +196,12 @@ func (e *ServiceSyncer) syncServiceAndEndpoints(ctx context.Context, fromService
 		return ctrl.Result{}, nil
 	}
 
+	// sync the loadbalancer status
+	if fromService.Spec.Type == corev1.ServiceTypeLoadBalancer && !apiequality.Semantic.DeepEqual(fromService.Status.LoadBalancer, corev1.LoadBalancerStatus{}) && !apiequality.Semantic.DeepEqual(fromService.Status.LoadBalancer, toService.Status.LoadBalancer) {
```

Comment:
I am curious to know why is part of the condition needed?
```
!apiequality.Semantic.DeepEqual(fromService.Status.LoadBalancer, corev1.LoadBalancerStatus{})
```

I would expect that if the state changes from having something in the `.Status.LoadBalancer` to having nothing, then we would like to have that reflected in the virtual Service resource too.

Reply:
ah, that makes sense!